new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/LoggingUtil.java

```diff
@@ -55,75 +83,142 @@ public static void collectLogs(Object itInstance, List namespaces) {
+      logger.warning(ex.getMessage());
+    }
 
     // get namespaces
-    for (var ns: Kubernetes.listNamespacesAsObjects().getItems()) {
-      if (namespace.equals(ns.getMetadata().getName())) {
-        writeToFile(ns, resultDir.toString(), namespace + "_ns.log");
+    try {
+      for (var ns : Kubernetes.listNamespacesAsObjects().getItems()) {
+        if (namespace.equals(ns.getMetadata().getName())) {
+          writeToFile(ns, resultDir, namespace + "_ns.log");
+        }
       }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
     }
 
     // get pvc
-    writeToFile(Kubernetes.listPersistentVolumeClaims(namespace), resultDir.toString(), namespace + "_pvc.log");
-
-    // get pv based on the weblogic.domainUID in pvc
-    for (var pvc : Kubernetes.listPersistentVolumeClaims(namespace).getItems()) {
-      if (pvc.getMetadata() != null
-          && pvc.getMetadata().getLabels() != null
-          && pvc.getMetadata().getLabels().get("weblogic.domainUID") != null) {
-        String label = pvc.getMetadata().getLabels().get("weblogic.domainUID");
-        writeToFile(Kubernetes.listPersistentVolumes(
-            String.format("weblogic.domainUID in (%s)", label)), resultDir.toString(), label + "_pv.log");
+    try {
+      writeToFile(Kubernetes.listPersistentVolumeClaims(namespace), resultDir, namespace + "_pvc.log");
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+    }
+
+    // get pv configuration and pv files based on the weblogic.domainUid label in pvc
+    try {
+      for (var pvc : Kubernetes.listPersistentVolumeClaims(namespace).getItems()) {
+        String label = Optional.ofNullable(pvc)
+            .map(metadata -> metadata.getMetadata())
+            .map(labels -> labels.getLabels())
+            .map(labels -> labels.get("weblogic.domainUid")).get();
+
+        // get the persistent volumes based on label weblogic.domainUid
+        V1PersistentVolumeList pvList = Kubernetes
+            .listPersistentVolumes(String.format("weblogic.domainUid = %s", label));
+        // write the persistent volume configurations to log
+        writeToFile(pvList, resultDir, label + "_pv.log");
+
+        // dump files stored in persistent volumes to
+        // RESULT_DIR/PVC_NAME/PV_NAME location
+        for (var pv : pvList.getItems()) {
+          String claimName = pvc.getMetadata().getName();
+          String pvName = pv.getMetadata().getName();
+          String hostPath = pv.getSpec().getHostPath().getPath();
+          try {
+            copyFromPV(namespace, hostPath,
+                Files.createDirectories(
+                    Paths.get(resultDir, claimName, pvName)));
+          } catch (ApiException apex) {
+            logger.warning(apex.getResponseBody());
+          } catch (Exception ex) {
+            ex.printStackTrace();
+            logger.warning(ex.getMessage());
+          }
+        }
+        logger.info("Done archiving the persistent volumes");
```

Comment:
include namespace, name, in the log message

Reply:
The copyDirectoryFromPod method  will print messages about the copy. The above logging is removed to be consistent with other write methods.